### PR TITLE
feat(registration): survey submission in own registrations

### LINKED ISSUE #9: Hente ut egne submissions som bruker
Som bruker ønsker jeg å se mine egne svar på spørreskjemaer, så jeg kan se hva jeg har sendt inn tidligere.

### PR DESCRIPTION
**Pull request by Eirsteir**
## Proposed changes

Add survey submission to registration serializer and implement ownership permsissions for regular users.

Issue number: closes #9 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [x] Build (`make PR`) was run locally without failures

### CODE REVIEW

File: app/content/models/registration.py [hunk 1]
```diff
@@ -237,3 +237,8 @@ def get_waiting_number(self):
                 if self == waiting:
                     return list(self.event.get_waiting_list()).index(self) + 1
         return None
+
+    @property
+    def submissions(self):
```
Comment (trthingnes) [inline app/content/models/registration.py:242]: This property is called `submissions`, but returns forms.

File: app/content/serializers/registration.py [hunk 2]
```diff
@@ -35,5 +38,9 @@ def get_user_info(self, obj):
             "allergy": user.allergy,
         }
 
+    def get_survey_submission(self, obj):
+        submissions = obj.submissions.filter(type=EventFormType.SURVEY)
```
Comment (trthingnes) [inline app/content/serializers/registration.py:42]: Should not user info also be used to retrieve survey submission?

**Review by trthingnes — CHANGES_REQUESTED** [review]

File: app/content/models/registration.py (hunk 1 above)
Comment (Eirsteir) [reply app/content/models/registration.py]: Very nicely spotted!

File: app/content/serializers/registration.py (hunk 2 above)
Comment (Eirsteir) [reply app/content/serializers/registration.py]: It's done in the `submissions` property which is now `get_submissions()` 😄

File: app/content/serializers/registration.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
 from rest_framework import serializers
 
 from app.common.serializers import BaseModelSerializer
+from app.forms.enums import EventFormType
+from app.forms.serializers.submission import SubmissionInRegistrationSerializer
 
 from ..models import Registration, User
```
Comment (olros) [inline app/content/serializers/registration.py:7]: Absolute import?

File: app/content/serializers/registration.py [hunk 4]
```diff
@@ -18,11 +21,11 @@ class Meta:
...
             "created_at",
+            "survey_submission",
             "waiting_number",
         ]
 
     def get_user_info(self, obj):
-        """ Gets the necessary info from user """
         user = User.objects.get(user_id=obj.user_id)
         return {
             "user_id": user.user_id,
```
Comment (olros) [inline app/content/serializers/registration.py:31]: Bytte ut med UserListSerializer for å standardisere litt kanskje

**Review by olros — APPROVED** [review]
Ser bra ut! 2 kommentarer bare på ting som egt er relevant her men allikevel kjekt å få endret 😉

**Pull request merged**